### PR TITLE
Upgrade Azure SDK and Jackson (#72833)

### LINKED ISSUE #66555: Hex numbers in YAML requests are treated as strings
**Elasticsearch version** (`bin/elasticsearch --version`):

```
{
  "number": "7.11.0-SNAPSHOT",
  "build_flavor": "default",
  "build_type": "docker",
  "build_hash": "00e58bef3ac3f0bd7ab68b90e38e11cac05fb49e",
  "build_date": "2020-12-15T14:29:43.737519Z",
  "build_snapshot": true,
  "lucene_version": "8.7.0",
  "minimum_wire_compatibility_version": "6.8.0",
  "minimum_index_compatibility_version": "6.0.0-beta1"
}
```

**Plugins installed**: []

**JVM version** (`java -version`):

```
{
  "version": "15.0.1",
  "vm_name": "OpenJDK 64-Bit Server VM",
  "vm_version": "15.0.1+9",
  "vm_vendor": "AdoptOpenJDK",
  "bundled_jdk": true,
  "using_bundled_jdk": true,
  "count": 1
}
```

**OS version** (`uname -a` if on a Unix-like system): Our official Docker container based on CentOS Linux 8

**Description of the problem including expected versus actual behavior**:

When Elasticsearch unmarshals YAML scalar values in hexadecimal format it interprets the value as a string, but it should be a number. For example `0x1` becomes `"0x01"` when it should a `1`. This causes problems when you expected the data type to be a number (like when reading the param value within painless).

Reference: https://yaml.org/spec/1.2/spec.html#id2766934

Relates: https://github.com/elastic/kibana/issues/85486

**Steps to reproduce**:

1. Create an ingest pipeline using `Content-Type: application/yaml`. Include a hexidecimal number.

```
curl \
  -u elastic:changeme \
  -H 'Content-Type: application/yaml' \
  -X PUT \
  "localhost:9200/_ingest/pipeline/yaml_test" \
  -d'---

description: YAML unmarshal bug demo
processors:
  - script:
      lang: painless
      params:
        number_base10: 1
        number_base16: 0x1
        string: hello
      source: >
        params.forEach((k, v) -> ctx[k] = v);'
```

2. Read the pipeline.

```
curl -u elastic:changeme -H 'Content-Type: application/yaml' -X GET "localhost:9200/_ingest/pipeline/yaml_test"
---
yaml_test:
  description: "YAML unmarshal bug demo"
  processors:
  - script:
      lang: "painless"
      params:
        number_base10: 1
        number_base16: "0x1"
        string: "hello"
      source: "params.forEach((k, v) -> ctx[k] = v);"
```

```
$ curl -u elastic:changeme -X GET "localhost:9200/_ingest/pipeline/yaml_test?pretty"
{
  "yaml_test" : {
    "description" : "YAML unmarshal bug demo",
    "processors" : [
      {
        "script" : {
          "lang" : "painless",
          "params" : {
            "number_base10" : 1,
            "number_base16" : "0x1",
            "string" : "hello"
          },
          "source" : "params.forEach((k, v) -> ctx[k] = v);"
        }
      }
    ]
  }
}
```

Note that `number_base16` is `"0x1"`. It should a 1.

### PR DESCRIPTION
**Pull request by rjernst**
This commit upgrades the Azure SDK to 12.11.0 and Jackson to 2.12.2. The
Jackson upgrade must happen at the same time due to Azure depending on
this new version of Jackson.

closes #66555
closes #67214

original PR #72833

### CODE REVIEW

**Comment by rjernst** [comment]
This PR is currently blocked on finding a way to not use the netty dns resolver. That (statically) tries to access /etc/hosts, which we do not allow (file reads outside of ES).

**Comment by fcofdez** [comment]
We were hitting this issue https://github.com/Azure/azure-sdk-for-java/pull/21163 in our tests.
```
diff --git a/plugins/repository-azure/build.gradle b/plugins/repository-azure/build.gradle
index be1e164cb54..6fca701f359 100644
--- a/plugins/repository-azure/build.gradle
+++ b/plugins/repository-azure/build.gradle
@@ -22,8 +22,8 @@ esplugin {
 }

 versions << [
-  'azure': '12.11.0',
-  'azureCore': '1.15.0',
+  'azure': '12.11.1',
+  'azureCore': '1.16.0',
   'azureCoreHttpNetty': '1.9.1',
   'azureAvro': '12.0.3',
```
With these versions we should be good to go.
Could you add me as a reviewer once the PR is ready?

**Comment by rjernst** [comment]
@fcofdez I updated as you suggested. There is still a failure in encrypted repository.

**Comment by rjernst** [comment]
Thanks @fcofdez, that worked. CI is passing now, can you take another look at the whole PR?

File: plugins/repository-azure/src/main/plugin-metadata/plugin-security.policy [hunk 1]
```diff
@@ -14,4 +14,6 @@ grant {
   // Used by jackson bean deserialization
   permission java.lang.RuntimePermission "accessDeclaredMembers";
   permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
+  // needed by netty dns resolver even though we don't use it (!)
+  permission java.io.FilePermission "/etc/hosts", "read";
```
Comment (fcofdez) [inline plugins/repository-azure/src/main/plugin-metadata/plugin-security.policy:18]: I think we need to add permissions to list and read files within `/etc/resolver`. See https://github.com/netty/netty/blob/d34212439068091bcec29a8fad4df82f0a82c638/resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java#L73

Comment (rjernst) [reply plugins/repository-azure/src/main/plugin-metadata/plugin-security.policy:18]: I added the persmissions and opened a followup (https://github.com/elastic/elasticsearch/issues/73479) to see how we might avoid using the netty dns resolver altogether.

**Review by fcofdez — APPROVED** [review]
LGTM 👍

**Pull request merged**